### ISSUE TITLE
Quickly scrolling through a long call stack crashes the UI

### ISSUE THREAD

**Issue by grouma** [flutter/devtools#2551]
I'm seeing the following exception thrown when using Dart DevTools built with DDC while scrolling quickly through a large call stack:

```
══╡ EXCEPTION CAUGHT BY SCHEDULER LIBRARY ╞═════════════════════════════════════════════════════════
js_primitives.dart:30 The following assertion was thrown during a scheduler callback:
js_primitives.dart:30 Assertion failed: org-dartlang-sdk:///flutter_web_sdk/lib/_engine/engine/html/picture.dart:54:10
js_primitives.dart:30 canvas == null || !_recycledCanvases.contains(canvas)
js_primitives.dart:30 is not true
js_primitives.dart:30 
js_primitives.dart:30 When the exception was thrown, this was the stack:
js_primitives.dart:30 dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 216:49  throw_
js_primitives.dart:30 dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 24:3    assertFailed
js_primitives.dart:30 lib/_engine/engine/html/picture.dart 54:62                                    _recycleCanvas
js_primitives.dart:30 lib/_engine/engine/html/picture.dart 316:7                                    [_applyPaint]
js_primitives.dart:30 lib/_engine/engine/html/picture.dart 576:7                                    update
js_primitives.dart:30 lib/_engine/engine/html/surface.dart 873:15                                   [_updateManyToOne]
js_primitives.dart:30 lib/_engine/engine/html/surface.dart 709:7                                    update
js_primitives.dart:30 lib/_engine/engine/html/clip.dart 329:11                                      update
js_primitives.dart:30 lib/_engine/engine/html/surface.dart 873:15                                   [_updateManyToOne]
js_primitives.dart:30 lib/_engine/engine/html/surface.dart 709:7                                    update
js_primitives.dart:30 lib/_engine/engine/html/offset.dart 49:11                                     update
js_primitives.dart:30 lib/_engine/engine/html/surface.dart 955:19                                   [_updateManyToMany]
js_primitives.dart:30 lib/_engine/engine/html/surface.dart 713:7                                    update
js_primitives.dart:30 lib/_engine/engine/html/clip.dart 108:11                                      update
```

**Comment by grouma** [comment]
Note this is likely a DDC issue. I can't repro on native nor Dart2JS.

cc @sigmundch

**Comment by grouma** [comment]
cc @yjbanov - might want to move this over to the Flutter SDK repository

**Comment by comstockz** [comment]
I've got the exact same issue right now.  Let me know if you figure it out.

**Comment by mkhtradm01** [comment]
I just got same error when I try to quickly pop navigation out of stack. The App freezes and this error comes up:

The following assertion was thrown during a scheduler callback:
Assertion failed: org-dartlang-sdk:///flutter_web_sdk/lib/_engine/engine/html/picture.dart:54:10
canvas == null || !_recycledCanvases.contains(canvas)
is not true

When the exception was thrown, this was the stack:
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 216:49  throw_
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 24:3    assertFailed
lib/_engine/engine/html/picture.dart 54:62                                    _recycleCanvas
lib/_engine/engine/html/picture.dart 321:9                                    [_applyPaint]
lib/_engine/engine/html/picture.dart 536:5                                    apply
lib/_engine/engine/html/surface.dart 367:5                                    build

**Comment by yjbanov** [comment]
We should definitely fix this, but I'm curious how DevTools managed to hit this issue. It should only affect the HTML renderer, and DevTools, AFAIK, uses CanvasKit.

**Comment by grouma** [comment]
I believe I was using CanvasKit. I was simply developing the app with `flutter run -d chrome`. @kenzieschmoll that should result in CanvasKit eh?

**Comment by mkhtradm01** [comment]
For me, I wasn’t using DevTools, I was running flutter run -d chrome for
web. On beta channel

**Comment by ferhatb** [comment]
Found root cause, will fix https://github.com/flutter/flutter/issues/72281 to resolve this.

**Comment by yjbanov** [comment]
For a little bit longer you'll need to pass `--web-renderer=canvaskit`. HTML is still the default.

**Comment by mkhtradm01** [comment]
> Found root cause, will fix [flutter/flutter#72281](https://github.com/flutter/flutter/issues/72281) to resolve this.

Thanks for that, looking forward for your effort 👍

**Comment by BeliliFahem** [comment]
I have same error today 😢  

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel beta, 1.25.0-8.2.pre, on macOS 11.1 20C69 darwin-x64, locale fr-FR)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 12.3)
[✓] Chrome - develop for the web
[✓] Android Studio
[✓] Android Studio
[✓] Android Studio (version 4.1)
[✓] IntelliJ IDEA Ultimate Edition (version 2020.3.1)
[✓] VS Code (version 1.52.1)
[✓] Connected device (1 available)

• No issues found!
```
